media/windows/ricoh-printer/ricoh_db_create.py

```diff
@@ -0,0 +1,122 @@
+#!/user/bin/env python3
+
+# Script which runs daily to update staff print totals
+# based on CSVs generated by the printer itself.
+
+import sys
+sys.path.insert(1, '../../python/')
+
+import csv
+import datetime
+import sqlite3
+from sqlite3 import Error
+
+#import ECC
+#import PDSChurch
+
+from pprint import pprint
+
+###########################################################
+
+ricoh_db = "ricoh.db"
+today = datetime.date.today()
+#datestring = f'{today.year}{today.month}{today.day}'
+debug_datestring = '20201013'
+datestring = debug_datestring
+
+def _strip(value):
+    #The CSV puts brackets around values, which we don't want
+    return value.strip('[').strip(']')
+
+def load_latest_csv(filename):
+
+    csv_rows = list()
+
+    with open(filename, encoding='utf-8') as csvfile: #TODO: change to filename when retrieval is automated
+        csvreader = csv.DictReader(csvfile)
+        for row in csvreader:
+            csv_rows.append(row)
+
+    print(f"== Loaded {len(csv_rows)} staffers from latest CSV")
+    return csv_rows
+
+def extract_csv_data(csv_rows):
+
+    def _extract_staffer(row, num, name):
+        return {
+            'num'           : num,
+            'name'          : name,
+            'total'         : int(row['Total Prints'].strip()),
+            'b&wtotal'      : int(row['B & W(Total Prints)'].strip()),
+            'colortotal'    : int(row['Color(Total Prints)'].strip()),
+        }
+
+    csv_staffers = list()
+
+    for row in csv_rows:
+        try:
+            num = _strip(row['User'])
+            name = _strip(row['Name'])
+        except ValueError as e:
+            print(e)
+            num = '000'
+            name = 'Other'
+        this_staffer = _extract_staffer(row, num, name)
+        csv_staffers.append(this_staffer)
+
+    print(f"== Extracted {len(csv_staffers)} staffers")
+    return csv_staffers
+
+
+def write_to_db(csv, ricoh_db): # Connects to SQLite db, or creates it if it doesn't exist
+
+    def  _insert_staffer(row, conn):
+        c = conn.cursor()
+        c.execute(f"""INSERT INTO 'printlog' (Date,User,Name,Total,BW,Color)
```

Comment:
I think you want to use the `?` notation to insert the values, like this example:

```python
c.execute("INSERT INTO 'printlog' (Date,User,Name,Total,BW,Color) VALUES (?,?,?,?,?,?)", (datestring, row['num'], ...etc.))
```

That's typed straight here into Github -- may not be 100% correct.  I *think* the double parentheses are necessary, though (i.e., the 2nd argument to `c.execute()` is a tuple).

Per https://docs.python.org/3/library/sqlite3.html:

> Usually your SQL operations will need to use values from Python variables. You shouldn’t assemble your query using Python’s string operations because doing so is insecure; it makes your program vulnerable to an SQL injection attack (see https://xkcd.com/327/ for humorous example of what can go wrong).
> 
> Instead, use the DB-API’s parameter substitution. Put ? as a placeholder wherever you want to use a value, and then provide a tuple of values as the second argument to the cursor’s execute() method. (Other database modules may use a different placeholder, such as %s or :1.) For example: ...example shown...